### PR TITLE
Add cjose build scripts

### PR DESCRIPTION
**Pull request by sergey-chernikov**

### CODE REVIEW

File: build_scripts/cjose_settings.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#
+#
+# ***********************************************************************************
+# * Copyright (C) 2020 - 2021, BlockSettle AB
```
Comment (Ation) [inline build_scripts/cjose_settings.py:4]: it is 2021

File: build_scripts/cjose_settings.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        return os.path.join(self._project_settings.get_common_build_dir(), 'cjose')
+
+    def is_archive(self):
+        return True
+
+    def config(self):
+        command = ['cmake',
+            self.get_source_dir(),
+            '-DOPENSSL_INCLUDE_DIR=' + os.path.join(self._project_settings.get_common_build_dir(), 'OpenSSL/include'),
+            '-DJANSSON_INCLUDE_DIR=' + os.path.join(self._project_settings.get_common_build_dir(), 'jansson/include'),
```
Comment (Ation) [inline build_scripts/cjose_settings.py:48]: Ideally it should be getting install dir directly from corresponding lib settings object. Qt and WS settings could be used as an example:
self.openssl = OpenSslSettings(settings)
os.path.join(self.openssl.get_install_dir(),'include'))

File: build_scripts/jansson_settings.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+from component_configurator import Configurator
+
+class JanssonSettings(Configurator):
+    def __init__(self, settings):
+        Configurator.__init__(self, settings)
+        self._version = '2.14'
+        self._script_revision = '1'
+        self._package_name = 'jansson-master'
+        self._package_url = 'https://github.com/akheron/jansson/archive/refs/heads/master.zip'
```
Comment (Ation) [inline build_scripts/jansson_settings.py:24]: please use specific revision

File: build_scripts/cjose_settings.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+from component_configurator import Configurator
+
+class CJOSE(Configurator):
+    def __init__(self, settings):
+        Configurator.__init__(self, settings)
+        self._version = '0.6.1'
+        self._script_revision = '1'
+        self._package_name = 'cjose-master'
+        self._package_url = 'https://github.com/sergey-chernikov/cjose/archive/refs/heads/master.zip'
```
Comment (Ation) [inline build_scripts/cjose_settings.py:24]: please use fixed revision

File: build_scripts/jansson_settings.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#
+#
+# ***********************************************************************************
+# * Copyright (C) 2020 - 2021, BlockSettle AB
```
Comment (Ation) [inline build_scripts/jansson_settings.py:4]: 2020

File: build_scripts/cjose_settings.py (hunk 4 above)
Comment (sergey-chernikov) [reply build_scripts/cjose_settings.py:24]: it's the only revision supporting cmake builds, and as you can see it's my fork, no other revisions are expected in the nearest future (the original lib is not updated for 6 years), I don't intend to create any revisions of it

File: build_scripts/jansson_settings.py (hunk 3 above)
Comment (sergey-chernikov) [reply build_scripts/jansson_settings.py:24]: I will if you point me out to how to get this link with revision of that lib from github, I can see a link only to master archive, and no releases tab. Though getting the latest master doesn't seem a bad idea except that we need to increase script revision to update the build.

File: build_scripts/cjose_settings.py (hunk 2 above)
Comment (sergey-chernikov) [reply build_scripts/cjose_settings.py]: Changed, thank you

File: build_scripts/cjose_settings.py (hunk 4 above)
Comment (Ation) [reply build_scripts/cjose_settings.py:26]: `self._version = '3f77bc82e078c6aa15cbffa796fb2d700612ecf4'
self._package_url = 'https://github.com/sergey-chernikov/cjose/archive/' + self._version + '.zip'
`

File: build_scripts/jansson_settings.py (hunk 3 above)
Comment (Ation) [reply build_scripts/jansson_settings.py:24]: self._package_url = 'https://github.com/akheron/jansson/archive/refs/tags/' + self._version + '.zip'

**Review by Ation — APPROVED** [review]

**Pull request merged**